drivers/segger/stream_rtt.c

```diff
@@ -0,0 +1,192 @@
+
+/****************************************************************************
+ * Name: rttstream_putc
+ ****************************************************************************/
+
+static void rttstream_putc(FAR struct lib_outstream_s *this, int ch)
+{
+  FAR struct lib_rttoutstream_s *stream =
+                                 (FAR struct lib_rttoutstream_s *)this;
+  stream->public.nput += SEGGER_RTT_PutChar(stream->channel, ch);
+}
+
+/****************************************************************************
+ * Name: rttstream_puts
+ ****************************************************************************/
+
+static int rttstream_puts(FAR struct lib_outstream_s *this,
+                          FAR const void *buf, int len)
+{
+  FAR struct lib_rttoutstream_s *stream =
+                                (FAR struct lib_rttoutstream_s *)this;
+  int ret = SEGGER_RTT_Write(stream->channel, buf, len);
+  stream->public.nput += ret;
+  return ret;
+}
+
+/****************************************************************************
+ * Name: rttstream_getc
+ ****************************************************************************/
+
+static int rttstream_getc(FAR struct lib_instream_s *this)
+{
+  FAR struct lib_rttinstream_s *stream =
+                                (FAR struct lib_rttinstream_s *)this;
+  int ch = -1;
+
+  DEBUGASSERT(stream);
+  stream->public.nget += SEGGER_RTT_Read(stream->channel, &ch, 1);
+  return ch;
+}
+
+/****************************************************************************
+ * Name: rttstream_gets
+ ****************************************************************************/
+
+static int rttstream_gets(FAR struct lib_instream_s *this,
+                          FAR void * buffer, int size)
+{
+  FAR struct lib_rttinstream_s *stream =
+                                (FAR struct lib_rttinstream_s *)this;
+  int ret;
+
+  DEBUGASSERT(stream);
+  ret = SEGGER_RTT_Read(stream->channel, buffer, size);
+  stream->public.nget += ret;
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lib_rttoutstream_open
+ *
+ * Description:
+ *   Initializes a stream for use with the configured RTT interface.
+ *
+ * Input Parameters:
+ *   stream - User allocated, uninitialized instance of struct
+ *            lib_rttoutstream_s to be initialized.
+ *   channel - SEGGER RTT channel number
+ *   bufsize - Size of the RTT buffer
+ *
+ * Returned Value:
+ *   None (User allocated instance initialized).
+ *
+ ****************************************************************************/
+
+void lib_rttoutstream_open(FAR struct lib_rttoutstream_s *stream,
+                           int channel, size_t bufsize)
```

Comment:
should we use SEGGER_BUFFER_SIZE_UP directly?

Reply:
or add below before line 122:
bufsize = bufsize ? bufsize : SEGGER_BUFFER_SIZE_UP;